spark/src/main/scala/org/apache/spark/sql/delta/commands/cdc/CDCReader.scala

```diff
@@ -193,6 +193,73 @@ object CDCReader extends CDCReaderImpl
     }
   }
 
+  /**
+   * A special BaseRelation wrapper for CDF reads for optimization at query planning stage.
+   */
+  case class DeltaCDFRelationAtAnalysis(
```

Comment:
copy of DeltaCDFRelation , without `with CatalystScan` / `override def buildScan/unhandledFilters`. 
new getScanDf for query plan generation